tests/status_response/test_java.py

```diff
@@ -100,5 +102,696 @@ class TestJavaStatusVersion(BaseStatusResponseTest):
+                            "1aee1ac99ceb6e2b8b0e1b1a0e3a4bde3a8aae59ba0e6ba8ce1af9de396b9e38085e6819ce482b8e18081e18290e0b88c"
+                            "e5ac9ce3a6b2e6a5abe1a39ce3a388e786a1e38c82e6b6a5e0b198e1a297d0b8e78ba6e0b6b9e0a08be18093e485a0e1a"
+                            "599e3a4b9e791a9e49b8ae4a6bde18cabe583a0e0b7ade1b599e382b6e789b2e6ab9ee19791e1ae93e18287e29787e4ae"
+                            "8ce19a98e38cb9e5a898e29685e7adb3d8b5e4b0ade5ae9be398b2e3888100e4a190e38cabe69a96e0b2ade1b399e39eb"
+                            "ae685b2e4ab8ee18685e7aca3e39ba6c2aee0ae8ce19c9c32e5b894e19781e68db3e19bb6e2b28ce1b299e1a283e2b8b2"
+                            "e5b1a0e0a384e28188e59a87e4aface5a19de398bae185b4e5b1a2e68384e185b1e28b93d786e4ae8ce1a098e3a0aee78"
+                            "9ace3a09ce28cabe79db7e6b98dc69ae19c98e398b2e6819cd080e6a2a8e49bb6e4b2ace5ae9ce392bae698bae6ab98e3"
+                            "a7a1e28cabe79db7e6b98de5b39ae1ae81e39cb000"
+                        )
+                    ).decode("utf-8"),
+                    "fmlNetworkVersion": 3,
+                    "mods": [],
+                    "truncated": True,
+                }
+            )
+        )
+        assert value is not None
+        return value
+
+
+@BaseStatusResponseTest.construct
+class TestForgeDataV1(TestJavaStatusResponse):
+    RAW = {
+        "description": {"text": "A Minecraft Server"},
+        "players": {"max": 20, "online": 0},
+        "version": {"name": "1.12.2", "protocol": 340},
+        "modinfo": {
+            "type": "FML",
+            "modList": [
+                {"modid": "minecraft", "version": "1.12.2"},
+                {"modid": "mcp", "version": "9.42"},
+                {"modid": "FML", "version": "8.0.99.99"},
+                {"modid": "forge", "version": "14.23.5.2859"},
+            ],
+        },
+    }
+
+    EXPECTED_VALUES = [
+        ("players", JavaStatusPlayers(0, 20, None)),
+        ("version", JavaStatusVersion("1.12.2", 340)),
+        ("motd", Motd(parsed=["A Minecraft Server", Formatting.RESET], raw={"text": "A Minecraft Server"}, bedrock=False)),
+        ("latency", 0),
+        ("raw", RAW),
+        (
+            "forge_data",
+            ForgeData(
+                fml_network_version=1,
+                channels=[],
+                mods=[
+                    ForgeDataMod(name="minecraft", marker="1.12.2"),
+                    ForgeDataMod(name="mcp", marker="9.42"),
+                    ForgeDataMod(name="FML", marker="8.0.99.99"),
+                    ForgeDataMod(name="forge", marker="14.23.5.2859"),
+                ],
+                truncated=False,
+            ),
+        ),
+    ]
+    OPTIONAL_FIELDS = [], {
+        "players": {"max": 20, "online": 0},
+        "version": {"name": "1.12.2", "protocol": 340},
+        "description": "A Minecraft Server",
+    }
```

Comment:
But subclassing from the existing test is such an easier solution

Reply:
How is that?

I designed tests to not fully support subclasses. They can only test that the value passed right, but they can't do parametrization on subclasses (which is the main benefit of this test system). For subclasses, we just create another `BaseStatusResponseTest` class which is responsible only for that subclass. It is how we do to `players`, `version` and other properties.

I lost this conversation somewhere, but you should subclass `BaseStatusResponseTest` directly (instead of `TestJavaStatusResponse`) and then in `build` method do only `ForgeData` class, not whole `JavaStatusResponse`.